node/src/node_configurator/mod.rs

```diff
@@ -131,13 +134,20 @@ pub fn data_directory_from_context(
                 .to_string();
             let right_local_data_dir =
                 wrong_local_data_dir.replace(&wrong_home_dir, &right_home_dir);
-            PathBuf::from(right_local_data_dir)
-                .join("MASQ")
-                .join(chain.rec().literal_identifier)
+            create_path_buf(chain, right_local_data_dir.to_string())
```

Comment:
Could have been written with less code. Look at the match as it has two arms only and both a lot of similar operations; while you call up your "create_path_buf(...)" in each arm, you could call it just once outside the match, because the match would always produce just the root path.